inference-engine/tests/functional/inference_engine/lp_transformations/fake_quantize_with_dynamic_intervals_transformation.cpp

```diff
@@ -0,0 +1,164 @@
+        testValue.outputHighConst;
+}
+
+typedef std::tuple<
+    ngraph::element::Type,
+    ngraph::Shape,
+    FakeQuantizeWithDynamicIntervalsTransformationTestValues> FakeQuantizeTransformationParams;
+
+class FakeQuantizeWithDynamicIntervalsTransformation : public LayerTransformation, public testing::WithParamInterface<FakeQuantizeTransformationParams> {
+public:
+    void SetUp() override {
+        const ngraph::element::Type precision = std::get<0>(GetParam());
+        const ngraph::Shape shape = std::get<1>(GetParam());
+        const FakeQuantizeWithDynamicIntervalsTransformationTestValues testValues = std::get<2>(GetParam());
+
+        actualFunction = get(precision, shape, testValues.inputLowConst, testValues.inpuHighConst, testValues.outputLowConst, testValues.outputHighConst);
+
+        SimpleLowPrecisionTransformer transform;
+        transform.add<ngraph::pass::low_precision::FakeQuantizeTransformation, ngraph::opset1::FakeQuantize>(testValues.params);
+        transform.transform(actualFunction);
+
+        referenceFunction = get(precision, shape, testValues.inputLowConst, testValues.inpuHighConst, testValues.outputLowConst, testValues.outputHighConst);
+    }
+
+    static std::string getTestCaseName(testing::TestParamInfo<FakeQuantizeTransformationParams> obj) {
+        ngraph::element::Type precision;
+        ngraph::Shape shape;
+        FakeQuantizeWithDynamicIntervalsTransformationTestValues testValues;
+        std::tie(precision, shape, testValues) = obj.param;
+
+        std::ostringstream result;
+        result << LayerTransformation::getTestCaseNameByParams(precision, shape, testValues.params) << testValues;
+        return result.str();
+    }
+
+private:
+    std::shared_ptr<ngraph::Node> makeTranspose(const ngraph::element::Type precision, const ngraph::Shape shape, const std::vector<float> low) {
+        const auto constant = std::make_shared<opset1::Constant>(precision, shape, low);
+        const auto transposeConstant = std::make_shared<opset1::Constant>(element::u32, Shape{ 4 }, std::vector<size_t>{1, 2, 3, 0});
+        const auto transpose = std::make_shared<opset1::Transpose>(constant, transposeConstant);
+        return transpose;
+    }
+
+    std::shared_ptr<ngraph::Node> makeConstant(const ngraph::element::Type precision, const ngraph::Shape shape, const std::vector<float> low) {
+        return std::make_shared<opset1::Constant>(precision, shape, low);;
+    }
+
+    std::shared_ptr<ngraph::Function> get(
+        ngraph::element::Type precision,
+        ngraph::Shape inputShape,
+        const bool inputLowConst,
+        const bool inpuHighConst,
+        const bool outputLowConst,
+        const bool outputHighConst) {
+        const auto input = std::make_shared<ngraph::opset1::Parameter>(precision, inputShape);
+        input->set_friendly_name("input");
+
+        const auto constantPresition = element::f32;
+        const auto constantShape = Shape{ 1, 1, 1, 1 };
```

Comment:
Are you sure that other Shapes will work correctly? What about per-channel quantization?

Reply:
It doesn't matter for this test. The tested source code doesn't use constant shape.